syntax = "proto3";

package hello;

service PaymentSystemService {
    rpc StartPayment (PaymentRequest) returns (PaymentResponse);
    rpc ConfirmPayment (PaymentConfirmation) returns (PaymentResponse);
}

message PaymentRequest {
    int64 payment_id = 1;
    int64 amount = 2;
    int64 credit_card = 3;
}

message PaymentResponse {
    bool success = 1;
}

message PaymentConfirmation {
    int64 payment_id = 1;
    bool perform_payment = 2;
}
